feat: add apertus model and cce (#3144)
OpenAccess-AI-Collective/axolotl: examples/apertus/README.md

> # Finetune Swiss-ai's Apertus with Axolotl

Reviewer: ```suggestion
# Finetune Swiss-AI's Apertus with Axolotl
```

> [Apertus](https://huggingface.co/collections/swiss-ai/apertus-llm-68b699e65415c231ace3b059) is a family of opensource models trained by Swiss-ai.
> 
> This guide shows how to fine-tune it with Axolotl with multi-turn conversations and proper masking.

Reviewer: ```suggestion
[Apertus](https://huggingface.co/collections/swiss-ai/apertus-llm-68b699e65415c231ace3b059) is a family of open-source models trained by Swiss-AI. This guide shows how to fine-tune these models with Axolotl with multi-turn conversations whilst ensuring correct masking between turns.
```

## Getting started

1. Install Axolotl following the [installation guide](https://docs.axolotl.ai/docs/installation.html). You need to install from main as Apertus is only on nightly or use our latest [Docker images](https://docs.axolotl.ai/docs/docker.html).

>     Here is an example of how to install from main for pip:

Reviewer: ```suggestion
Here is an example of how to install from main for pip:
```

```bash
# Ensure you have Pytorch installed (Pytorch 2.6.0 min)
git clone https://github.com/axolotl-ai-cloud/axolotl.git
cd axolotl

pip3 install packaging==23.2 setuptools==75.8.0 wheel ninja
pip3 install --no-build-isolation -e '.[flash-attn]'

# Install CCE https://docs.axolotl.ai/docs/custom_integrations.html#cut-cross-entropy
python scripts/cutcrossentropy_install.py | sh
```

2. (Optional, highly recommended) Install XIELU CUDA

```bash
## Recommended for reduced VRAM and faster speeds

# Point to CUDA toolkit directory
# For those using our Docker image, use the below path.
export CUDA_HOME=/usr/local/cuda

pip3 install git+https://github.com/nickjbrowning/XIELU@59d6031 --no-build-isolation --no-deps
```

For any installation errors, see [XIELU Installation Issues](#xielu-installation-issues)

3. Run the finetuning example:

```bash
axolotl train examples/apertus/apertus-8b-qlora.yaml
```

This config uses about 8.7 GiB VRAM.

Let us know how it goes. Happy finetuning! 🚀

> ### TIPS

Reviewer: ```suggestion
### Tips
```

- For inference, the official Apertus team recommends `top_p=0.9` and `temperature=0.8`.

> - You can run a full finetuning by removing the `adapter: qlora` and `load_in_4bit: true` from the config.

Reviewer: ```suggestion
- You can instead use full paremter fine-tuning by removing the `adapter: qlora` and `load_in_4bit: true` from the config.
```
